tools/src/main/java/org/apache/kafka/tools/ProducerPerformance.java

```diff
@@ -506,6 +584,9 @@ public ConfigPostProcessor(ArgumentParser parser, String[] args) throws IOExcept
             if (numRecords != null && numRecords <= 0) {
                 throw new ArgumentParserException("--num-records should be greater than zero", parser);
             }
+            if (warmupRecords != null && warmupRecords >= numRecords) {
```

Comment:
`warmupRecords != null` is unnecessary. Also, could you please change the type of `numRecords` and `warmupRecords` from `Long` to `long`?